modules/kubernetes-addons/crossplane/main.tf

```diff
@@ -14,5 +14,46 @@ module "helm_addon" {
   helm_config       = local.helm_config
   irsa_config       = null
 
-  depends_on = [kubernetes_namespace_v1.irsa]
+  depends_on = [kubernetes_namespace_v1.crossplane]
+}
+
+resource "kubectl_manifest" "aws_provider" {
+  for_each  = toset(data.kubectl_path_documents.aws_provider.documents)
+  yaml_body = each.value
+
+  depends_on = [module.helm_addon]
+}
+
+module "aws_provider_irsa" {
+  source                            = "../../../modules/irsa"
+  eks_cluster_id                    = var.eks_cluster_id
+  create_kubernetes_namespace       = false
+  create_kubernetes_service_account = false
+  kubernetes_namespace              = local.namespace
+  kubernetes_service_account        = "provider-aws-*"
```

Comment:
service account name here has Asterisk sign, is that being replaced automatically? if not then the name is invalid

Reply:
Crossplane creates a random service account which we don't have control. So, their recommendation is to use "provider-aws-*". `*` is basically a random string

We can add this trust relationship with `StringLike`. You can see the change below in IRSA data sources